tests/functional/restart/55-remote-init-not-required.t

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
```

Comment:
Not sure this test is testing the fix. 
I am getting it passing with the fix commented out, and passing on master.

Reply:
Odd, bc it fails for me on master...

Reply:
Also tried commenting out the fix, the test still passes for me. Maybe config related or non-deterministic, I'll try your global.cylc and report back.

Reply:
I ran the test without the fix on GH Actions here, and it failed: https://github.com/MetRonnie/cylc-flow/runs/1830200176?check_suite_focus=true

Reply:
I can not get this to fail (running multiple times), either on master or on this branch with the fix commented out. But @wxtim is getting it to fail (I have tried using Tim's global-tests.cylc config). Would ideally like to get this to fail, if @MetRonnie can confirm it fails without the fix then I guess it is environment related for me.   
![image](https://user-images.githubusercontent.com/37735232/106887027-7ebbb080-66dc-11eb-8b34-14d25fb7172e.png)

Reply:
> I ran the test without the fix on GH Actions here, and it failed: https://github.com/MetRonnie/cylc-flow/runs/1830200176?check_suite_focus=true

Yes, must be environment related, presumably you are getting it failing locally too? Maybe ssh errors are being suppressed in my environment, not sure why though. Will try a fresh conda install and see if that helps!

Reply:
To conclude: this was because Mel had previously ssh'd into localhost, so it didn't prompt her, whereas Tim and I have not previously done so, so we were being promted to confirm the authenticity of the fingerprint etc. That is why Tim and I (and a freshly spun up GH Actions VM) were getting the ssh error. Going to try and rethink the test

Reply:
@wxtim @datamel I've replaced the functional test with a unit test now, involves a bit of awkward mocking as ever